added plugin for smallest.ai tts models

hamees-sayed opened the pull request:
I have added support for the non streaming and non auto regressive model by [smallest.ai](https://smallest.ai), following the existing TTS structure in adjacent plugins.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/livekit/agents?pullRequest=890) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/livekit/agents?pullRequest=890) before we can accept your contribution.<br/>**12** out of **14** committers have signed the CLA.<br/><br/>:white_check_mark: longcw<br/>:white_check_mark: Antonyesk601<br/>:white_check_mark: CathyL0<br/>:white_check_mark: levity<br/>:white_check_mark: dan-ince-aai<br/>:white_check_mark: JohnBurtt10-bot<br/>:white_check_mark: alexshelkov<br/>:white_check_mark: lucasolinas<br/>:white_check_mark: jayeshp19<br/>:white_check_mark: yaniv-peretz<br/>:white_check_mark: hamees-sayed<br/>:white_check_mark: thavidu<br/>:x: Stefano Scotto<br/>:x: mfernandez-gladia<br/><hr/>**Stefano Scotto** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/livekit/agents?pullRequest=890) it.</sub>

theomonnom commented:
Hey, thanks for your PR, do you think we could get an API key so we can add this new TTS to our tests?

hamees-sayed commented:
Hey @theomonnom you can get the api key from here https://waves.smallest.ai/apikeys

hamees-sayed commented:
Hi @theomonnom! Just checking if you had a chance to add smallest's TTS to the tests?

hamees-sayed commented:
the error seems to be failing because of missing elevanlabs tts key?

prashantmetadome commented:
@hamees-sayed Thank you for your PR. I integrated it in my local fork of livekit but I am getting the following error in the integration of smallest ai TTS. Kindly help. 

`samples = np.frombuffer(self._initial_buffer, dtype=np.int16)
ValueError: buffer size must be a multiple of element size`

Note: It was working till day before yesterday.

hamees-sayed commented:
@prashantmetadome the new changes should fix it
@theomonnom any status on when this could be merged or if I have to make any changes?

theomonnom commented:
I'll create a ticket internally for it, thanks!

hamees-sayed commented:
hey @theomonnom can you please approve the workflow, i have fixed the ruff formatting issue

hamees-sayed commented:
i am so sorry for pinging you so much @theomonnom but i think now it should do it i have also smallest in readme and workflow

theomonnom commented:
> i am so sorry for pinging you so much @theomonnom but i think now it should do it i have also smallest in readme and workflow

Nice thank you, we'll create a key and add it to our secrets before merging

hamees-sayed commented:
Hey @theomonnom I have tested the pytest module and github workflow and everything's passing on my end, can you please approve the workflow here :)

hamees-sayed commented:
@theomonnom hi any update on this PR?

hamees-sayed commented:
@theomonnom the PR has been stale for a while can you please have a look?

longcw committed:
wait for video track from avatar plugins (#2627)

longcw committed:
add http_options for gemini LLM and realtime model (#2640)

david-rodriguez committed:
correctly passing speaking_rate to StreamingAudioConfig (#2631)

CyprienRicqueB2L committed:
Fix : Increase audio mixer timeout (#2646)

raghavjaistra committed:
handling multiple audio chunk output (#2641)

bcherry committed:
Fix Hume TTS (#2639)

Co-authored-by: David Zhao <dz@livekit.io>

bcherry committed:
Update sarvam defaults, add 2.5 (#2618)

longcw committed:
fix tracing param in openai realtime (#2652)

longcw committed:
raise error from gladia stt for fallback adapter and retry (#2653)

longcw committed:
fix await tasks groups never return (#2654)

davidzhao committed:
chore: add note for job_context.api usage (#2655)

davidzhao committed:
fix(google): update dependency versions (#2658)

davidzhao committed:
feat(baseten): add LLM module (#2657)

longcw committed:
cleanup tee in agent activity (#2660)

theomonnom committed:
fix duplicated audio on flush (#2663)

longcw committed:
fix transcription sync warning when gemini no text output (#2661)

theomonnom committed:
introduce test & eval primitives (#2662)

theomonnom committed:
livekit-agents v1.1.2 (#2664)

theomonnom committed:
add --ignore-changesets to update_versions.py (#2665)

longcw committed:
remove frame_size_ms param when creating AudioStream (#2667)

theomonnom committed:
livekit-agents v1.1.3 (#2668)

mfernandez-gladia committed:
Gladia STT - add new parameters to gladia stt (#2649)

longcw committed:
expose automatic_function_calling config for google LLM (#2675)

longcw committed:
start user away timer after user join (#2676)

Panmax committed:
preserve created_at timestamp when updating instructions (#2677)

theomonnom committed:
evals: add maybe_* utils  (#2681)

theomonnom committed:
evals: better error message for assertions  (#2682)

longcw committed:
use parameters_json_schema for raw function tool with google LLM (#2686)

longcw committed:
import TextInputEvent from room_io (#2679)

longcw committed:
reset agent and user state after session closed (#2691)

theomonnom committed:
evals: RunResult final_output on Agent tasks  (#2696)

theomonnom committed:
evals: AgentTask GetEmailAdress tests e.g (#2697)

theomonnom committed:
allow optional RunResult output_type (#2698)

theomonnom committed:
evals: add EventRangeAssert utils  (#2699)

bcherry committed:
Add hedra extra (#2705)

longcw committed:
add markdown filter for tts and transcription nodes (#2695)

toubatbrian committed:
Fix Example Typo (#2706)

davidzhao committed:
Inworld TTS (#2693)

Co-authored-by: Michael Solati <mkslt04@gmail.com>
Co-authored-by: Long Chen <longch1024@gmail.com>

longcw committed:
add warning for deprecated speed and emotion control for cartesia tts (#2708)

MichaelSolati committed:
fix(plugins-inworld): change default voice to Ashley (#2707)

theomonnom committed:
deepgram: disable smart_format by default (#2704)

theomonnom committed:
livekit-agents 1.1.4 (#2709)

arpesenti committed:
Preserve original path when connecting to web socket (fix for #2700) (#2702)

longcw committed:
disconnect room when session closed due to participant disconnected (#2712)

longcw committed:
make sure audio_output.flush called when capture frame failed (#2718)

ShayneP committed:
Update Inworld README (#2723)

htrivedi99 committed:
Updating whisper API (#2726)

Co-authored-by: David Zhao <dz@livekit.io>

simplegr33n committed:
Lock google-genai package to stable v1.20.0 (#2725)

davidzhao committed:
fix(google): pass in raw schema according to genai 1.20 spec (#2727)

mrkowalski committed:
feat(google): expose seed parameter in LLM.chat (#2721)

theomonnom committed:
add front-desk agent example (#2724)

theomonnom committed:
fix InlineAgent agent resume on error (#2730)

theomonnom committed:
add ChatContext.merge & merge inline tasks chat_ctx (#2731)

theomonnom committed:
better GetEmailAgent instructions  (#2732)

theomonnom committed:
exclude function_call inside ChatContext.merge (#2733)

longcw committed:
upgrade google genai to 1.23 (#2743)

longcw committed:
support 11labs auto mode with sentence tokenizer (#2744)

theomonnom committed:
add livekit-blingfire (#2734)

theomonnom committed:
remove changesets (#2749)

theomonnom committed:
uv: ignore blingfire  (#2750)

theomonnom committed:
fix aggregate-dumps when no file is present (#2751)

theomonnom committed:
run tts tests on top10 providers (#2752)

theomonnom committed:
delete changesets x2 (#2753)

theomonnom committed:
add build CI  (#2754)

theomonnom committed:
fix blingfire build CI (#2756)

theomonnom committed:
BlingFire: use Release config on Windows  (#2757)

theomonnom committed:
build blingfire for macos x86 & linux arm64 (#2758)

BumaldaOverTheWater94 committed:
Nova Sonic Realtime Plugin (#2740)

theomonnom committed:
keep aws nova sonic optional (#2760)

theomonnom committed:
livekit-agents 1.1.5 (#2761)

zachoverflow committed:
Fix LMNT plugin docs (#2762)

bcherry committed:
Update new plugin readmes for format and links (#2571)

BumaldaOverTheWater94 committed:
fix update_chat_ctx bug (#2763)

dkeller-sondermind committed:
Include item id when converting to LG messages (#2767)

longcw committed:
fix schedule speech on windows when monotonic_ns resolution is rough (#2770)

theomonnom committed:
Merge branch 'main' into theo/agents1.2

theomonnom committed:
add Blingfire tokenizer & use it by default  (#2771)

theomonnom committed:
fix RealtimeModel generate_reply authorization (#2773)

longcw committed:
support aligned transcripts from tts (#2580)

bcherry committed:
Install optional dependencies during docs gen (#2766)

longcw committed:
ignore empty sentence in tts stream adapter (#2777)

longcw committed:
Merge remote-tracking branch 'origin/main' into theo/agents1.2

longcw committed:
fix types for agents 1.2 (#2778)

longcw committed:
fix MockTools type (#2781)

theomonnom committed:
fix RunResult order of fnc_call & agent_handoff (#2782)

theomonnom committed:
fix types (#2783)

theomonnom committed:
fix tr_input  (#2784)

theomonnom committed:
fix GetEmailAgent instructions (#2786)

longcw committed:
fix blingfire tokenizer test (#2785)

longcw committed:
support tts with realtime model (audio in, text out) (#2628)

theomonnom committed:
fix assistant message order on the RunResult (#2787)

theomonnom committed:
fix FrontDeskAgent list_available_slots (#2788)

fabitokki committed:
add livekit-plugins-mistralai (#2772)

Co-authored-by: Théo Monnom <theo.monnom@outlook.com>

theomonnom committed:
fix docker-compose typo (#2789)

theomonnom committed:
initial evals for the FrontDesk agent  (#2790)

theomonnom committed:
ignore empty assistant messages  (#2792)

longcw committed:
suppress main_stream ended error in stt fallback adapter (#2684)

theomonnom committed:
evals: add CI (#2791)

theomonnom committed:
evals ci: use python 3.12 (#2793)

theomonnom committed:
fix confirmation/validation ambiguity on GetEmailAgent instructions  (#2794)

theomonnom committed:
fix GetEmailAgent ambiguity x2

Is44m committed:
fix Orus voice name definition (#2797)

theomonnom committed:
fix GetEmailAgent ambiguity x3

theomonnom committed:
fix DriveThruAgent evals ambiguity

jeradf committed:
punctuation free turn detector (#2717)

Co-authored-by: David Zhao <dz@livekit.io>

longcw committed:
fix aws sonic type checking (#2804)

longcw committed:
fix deepgram stt docs (#2803)

htrivedi99 committed:
Hotfix for Baseten STT (#2801)

theomonnom committed:
frontdesk: ToolError example  (#2808)

theomonnom committed:
fix inactive user instructions (#2809)

theomonnom committed:
fix import error

theomonnom committed:
fix BackgroundAudio hanging on close error  (#2814)

longcw committed:
reset closing_ws for openai stt (#2813)

theomonnom committed:
avoid sid error in console mode (#2815)

theomonnom committed:
ignore livekit api when using console mode  (#2816)

CyprienRicqueB2L committed:
add stream_timeout_ms to BackgroundAudioPlayer (#2796)

longcw committed:
fix FunctionToolsExecutedEvent import (#2832)

davidzhao committed:
feat: ability to use remote EOT inference when deployed in Cloud (#2780)

kechako committed:
Add support for CustomPronunciations in Google TTS plugin (#2692)

davidzhao committed:
tweaks to drive-thru test to make it mostly compatible with gemini flash 2.0

theomonnom committed:
evals API improvements (#2846)

theomonnom committed:
make arguments optional for mock_tools (#2847)

theomonnom committed:
allow returning Exception inside function tools (#2848)

theomonnom committed:
add envvar to enable verbose evals logs (#2849)

BumaldaOverTheWater94 committed:
Nova Sonic example (#2817)

donalffons committed:
Prevent console mode from crashing (#2853)

longcw committed:
preemptive generation before end of user turn (#2728)

Co-authored-by: Théo Monnom <theo.8bits@gmail.com>

theomonnom committed:
fix no speech_id argument for _update_agent_state

theomonnom committed:
fix next_event return type (#2856)

theomonnom committed:
evals: add docstrings to the public API (#2857)

theomonnom committed:
only print the judge result when verbose is enabled (#2858)

theomonnom committed:
better preemptive_generation docstrings

bcherry committed:
add contains_agent_handoff assertion (#2862)

theomonnom committed:
allow editing SpeechHandle allow_interruptions & add RunContext.disallow_interruptions (#2864)

kath0la committed:
Small fix to README (#2861)

eliooooooot committed:
use synchronized transcript for interrupted session.say() responses (#2843)

Co-authored-by: Théo Monnom <theo.monnom@outlook.com>

theomonnom committed:
fix evals test (#2865)

theomonnom committed:
Merge branch 'main' into theo/agents1.2

theomonnom committed:
fix aws sonic tools (#2859)

Co-authored-by: BumaldaOverTheWater94 <83429948+BumaldaOverTheWater94@users.noreply.github.com>

theomonnom committed:
log metrics in extra (#2868)

BumaldaOverTheWater94 committed:
accidentally omit a docstring (#2869)

theomonnom committed:
livekit-agents v1.1.6 (#2872)

Panmax committed:
fix log extra field handling in log.py (#2875)

longcw committed:
fix aws realtime model types (#2877)

longcw committed:
fix ruff and types (#2889)

longcw committed:
add opentelemetry trace (#2873)

Co-authored-by: Théo Monnom <theo.8bits@gmail.com>

theomonnom committed:
fix unordered user messages (#2891)

theomonnom committed:
fix livekit-agents 1.2 tests (#2866)

theomonnom committed:
cleanup & prepare for release (#2893)

theomonnom committed:
Merge branch 'main' into theo/agents1.2

davidzhao committed:
chore: export PlayHandle type (#2903)

this is pretty useful to keep track as it's returned by BackgroundAudio.play

also bumping api dep version

longcw committed:
fix gemini realtime user transcription sent twice (#2899)

BumaldaOverTheWater94 committed:
append framework ID to User-Agent Header (#2896)

theomonnom committed:
add prometheus support (#2908)

theomonnom committed:
better metrics name

longcw committed:
add gen_ai attributes to llm_request (#2905)

Co-authored-by: Dan McAulay <dmcaulay@gmail.com>

longcw committed:
fix types and aws realtime model (#2910)

longcw committed:
add gemini tts (beta) (#2834)

theomonnom committed:
fix DatastreamIO cancellation race (#2911)

theomonnom committed:
DataStreamIO wait for start when capturing_frame (#2912)

theomonnom committed:
livekit-agents v1.1.7 (#2913)

longcw committed:
fix TTS fallback adapter metrics_collected event (#2890)

longcw committed:
add model property for llm plugins (#2914)

theomonnom committed:
nit: improve drivethru (#2918)

sascotto committed:
removing ctx.connect() from examples (#2909)

longcw committed:
expose tokenizer option for cartesia tts (#2916)

theomonnom committed:
remove openai prewarm (#2919)

Panmax committed:
add tts_audio_duration to usage metrics collection (#2915)

alfredguiaugment committed:
add inference process health check endpoint (#2906)

theomonnom committed:
only check inference process health if started (#2920)

davidzhao committed:
fix missing field in UsageCollector (#2929)

theomonnom committed:
Merge branch 'main' into theo/agents1.2

theomonnom committed:
rename prometheus metrics

theomonnom committed:
ruff

theomonnom committed:
livekit-agents 1.2.0 (#2482)

theomonnom committed:
fix ruff & bump livekit-agents 1.2.0 (#2936)

theomonnom committed:
livekit-agents 1.2.1 (#2937)

longcw committed:
fix cartesia non-streaming tts (#2942)

theomonnom committed:
add RecorderIO and --record flag to the console mode (#2934)

mateuszkulpa committed:
chore: remove prometheus database from repository (#2944)

levity committed:
parameterize inference worker init timeout (#2805)

mike-r-mclaughlin committed:
openai: add support for service_tier (#2945)

CathyL0 committed:
fix: upgrade bithuman library to unblock accessing agents (#2948)

longcw committed:
fix duplicated user messages when preemptive generation canceled (#2949)

longcw committed:
fix azure stt update options and add logs for error reason (#2954)

ladvoc committed:
[🤖 readme-manager] Update README (#2956)

Commit:
Explictly calling ctx.connect before wait_for_participant (#2957)

longcw committed:
azure stt: disable language detection if only one language sepcified (#2959)

longcw committed:
gemini: emit input_speech_started when new generation created (#2963)

theomonnom committed:
evals: fix realtime model RuntimeError (#2965)

johncDepop committed:
missing `f` on logs (#2964)

theomonnom committed:
add RunContext.wait_for_playout and guard against deadlocks (#2966)

bml1g12 committed:
feat(realtime_model): correctly emit errors when the response is done (#2967)

theomonnom committed:
slightly optimize import time (#2968)

theomonnom committed:
increase RoomInput frame_size_ms to 50ms (#2970)

longcw committed:
add warning when enabling unprovided input/output sinks (#2969)

davidzhao committed:
Handle RN format for preconnect mimeType (#2952)

Co-authored-by: davidliu <davidliu@deviange.net>

longcw committed:
tune vad min_silence_duration and min_endpointing_delay (#2953)

karlson-anam committed:
feat: add anam avatar (#2938)

Co-authored-by: karlson lee <karlsonlee@karlsons-MacBook-Pro.local>
Co-authored-by: Long Chen <longch1024@gmail.com>

longcw committed:
fix types for anam avatar plugin (#2976)

longcw committed:
fix 11labs tts when audio is an empty string (#2973)

longcw committed:
support resume agent from a false interruption (#2852)

Antonyesk601 committed:
feat: add simli avatar with example (#2923)

longcw committed:
add simli plugin to ci (#2978)

theomonnom committed:
remove Resemble from CI  (#2979)

longcw committed:
clean up avatar example and add retry for datastream io rpc call (#2943)

longcw committed:
expose transcription sync speed to RoomOutputOptions (#2984)

longcw committed:
hume tts: raise error message from the api (#2982)

theomonnom committed:
io: add input source hierarchy & cleanup (#2983)

theomonnom committed:
fix AgentFalseInterruptedEvent none message (#2987)

theomonnom committed:
rename AgentFalseInterruptedEvent -> AgentFalseInterruptionEvent (#2988)

theomonnom committed:
nit: update AgentFalseInterruptionEvent (#2989)

theomonnom committed:
fix deadlock & session close race (#2997)

longcw committed:
wait on_exit before pause scheduling (#2996)

mfernandez-gladia committed:
Gladia STT - add region parameter to gladia stt (#2995)

levity committed:
improve chat_cli status printing (#2991)

CathyL0 committed:
fix: upgrade bithuman library version (#2998)

theomonnom committed:
Revert "Improve chat_cli status printing" (#2999)

theomonnom committed:
improve GetEmailTask instructions (#3002)

theomonnom committed:
message should be None when empty (#3003)

theomonnom committed:
ci: enable verbose evals (#3004)

theomonnom committed:
fix sensitive TTS tests (#3005)

theomonnom committed:
livekit-agents 1.2.2 (#3006)

dan-ince-aai committed:
chore(assemblyai): update termination message (#3001)

longcw committed:
fix gemini realtime generate_reply (#3017)

longcw committed:
pin the hedra avatar sample rate to 16k (#3027)

longcw committed:
add metadata when set tracer provider (#3034)

alexshelkov committed:
Fix incorrect streaming flag for neuphonic (#3036)

mfernandez-gladia committed:
Gladia STT - fix region parameter to gladia stt on query parameters (#3018)

JohnBurtt10-bot committed:
feat(azure-stt): add phrase list support for keyword boosting (#3024)

longcw committed:
fix aws stt restarting (#2994)

longcw committed:
improve tts error trace and expose cartesia error message (#3028)

longcw committed:
add retry_on_chunk_sent to FallbackAdapter for allowing retry after chunk sent (#3033)

jayeshp19 committed:
Fix RuntimeError on scheduling speech and on mark_generation_done (#3032)

Co-authored-by: Théo Monnom <theo.monnom@outlook.com>

longcw committed:
add modalities arg to OAI realtime model with_azure (#3049)

yaniv-peretz committed:
enable - gemini-live-2.5-flash-preview (#3016)

lucasolinas committed:
Update events.py with new AWS Nova Sonic voice (#3048)

Commit:
Updating minimal worker to connect to room (#3065)

thavidu committed:
Fixing chunk size bug on inworld plugin-- matches hume and neuphonic (#3067)

davidzhao committed:
feat: use default load fnc and thresholds when hosting on Cloud (#3069)

longcw committed:
fix: call on_attached or on_detached when set agent input and output (#3063)

longcw committed:
fix using AgentTask in on_enter and on_ext (#3051)

longcw committed:
use tts as fallback when realtime model returns text response (#3050)

longcw committed:
add tts stream pacer to generate speech lazily (#3074)

theomonnom committed:
livekit-agents v1.2.3 (#3077)

longcw committed:
google stt: add enable_word_time_offsets option (#3079)

CathyL0 committed:
feat: support bithuman cloud avatar (#3055)

Co-authored-by: Long Chen <longch1024@gmail.com>

hamees-sayed committed:
smallest plugin added

hamees-sayed committed:
Merge branch 'livekit:main' into smallest-plugin

hamees-sayed committed:
remove chunk limit
